sql/src/main/java/io/crate/jobs/PageBucketReceiver.java

```diff
@@ -56,6 +56,14 @@
  */
 public interface PageBucketReceiver {
 
+    /**
+     * Receives a bucket from an upstream which holds result data. This method may
```

Comment:
I think the contract is as follows:

This is called *once* per bucketIdx initially; then exactly once for each `bucketIdx/pageResultListener.needMore=true`

(The important difference to what is currently documented in this PR is that it is *not* expected to get another call if `needMore=false` for that bucketIdx)

Reply:
Thanks, I've rephrased this to reflect your suggestion.